sema4ai/src/sema4ai_code/robocorp_language_server.py

```diff
@@ -1083,6 +1083,47 @@ def _is_port_open(port) -> bool:
 
         return None
 
+    def _get_actions_metadata(
+        self,
+        action_package_path: str,
+        monitor: IMonitor,
+    ) -> ActionResultDict:
+        from sema4ai_code.robo.collect_actions import get_metadata
+
+        p = Path(uris.to_fs_path(action_package_path))
+        if not p.exists():
+            msg = f"Unable to collect actions metadata from: {p} because it does not exist."
+            return dict(success=False, message=msg, result=None)
+
+        try:
+            if not self.workspace:
+                return dict(
+                    success=False, message="No workspace currently open", result=None
+                )
+
+            result = get_metadata(self._pm, action_package_path, monitor)
+            if not result.success:
```

Comment:
I think you don't need the `if not result.success:`... it can always return `result.as_dict()` here (no need for the other `return dict(success=True, message=None, result=result.result)`).